src/gen-mapping.ts

```diff
@@ -216,7 +230,11 @@ export class GenMapping {
 
     setSourceContent = (map, source, content) => {
       const { _sources: sources, _sourcesContent: sourcesContent } = map;
-      sourcesContent[put(sources, source)] = content;
+
+      const sourceIndex = get(sources, source);
+      if (sourceIndex !== undefined) {
+        sourcesContent[sourceIndex] = content;
+      }
```

Comment:
While not directly related to `applySourceMap` I've seen while preparing my PR to postcss that `setSourceContent` accepts to add any source content even for unknown sources.

However this would break if the `setSourceContent` is called before adding mappings and I don't know if that's needed or not.

`source-map-js` has a different approach where they store all source contents by `Map<path, content>` and create the `sourcesContent` when transforming to an encoded map, effectively dropping the unneeded sources.

What do you think is the best approach?

Reply:
> However this would break if the setSourceContent is called before adding mappings and I don't know if that's needed or not.

Can you describe what you mean? `setSourceContent` should create a new index, and if a mapping is ever added with that source filename, that index will be used. Sourcemaps mappings don't need to be monotonic to the source locations.

> source-map-js has a different approach where they store all source contents by Map<path, content> and create the sourcesContent when transforming to an encoded map, effectively dropping the unneeded sources.

I think this depends on the answer to the previous question. For the general case, I think it's fine to add source content for sources that never receive a mapping (and it keeps the library fast). But I'm not sure of a situation that your code will do that.